### PR TITLE
Lazy nozzle

### PR DESCRIPTION
**Pull request by MTCam**
This builds on @kaushikcfd's work to make a lazy nozzle.  Just getting it working with the current code from y1-production.

cc: @kaushikcfd, @anderson2981

### CODE REVIEW

**Comment by MTCam** [comment]
This supersedes both #7 and #9.

**Comment by kaushikcfd** [comment]
Thanks for the PR!

> This supersedes #7 

Is the plan here to cherry-pick the commits from #7, as some array ops here are still not compatible with all array contexts.

**Comment by majosm** [comment]
> This supersedes both #7 and #9.

Any reason we can't just merge #9? Might be nice to get that stuff in before piling lazy on top of it.

**Comment by MTCam** [comment]
> > This supersedes both #7 and #9.
> 
> Any reason we can't just merge #9? Might be nice to get that stuff in before piling lazy on top of it.

Just waiting on @anderson2981 to merge stuff.

**Comment by MTCam** [comment]
> Thanks for the PR!
> 
> > This supersedes #7
> 
> Is the plan here to cherry-pick the commits from #7, as some array ops here are still not compatible with all array contexts.

I don't think I understand the concern.   Should we concern ourselves with that before we are ready to use all array contexts?

**Comment by kaushikcfd** [comment]
> Should we concern ourselves with that before we are ready to use all array contexts?

If this PR supersedes #7, then I think yes.

**Comment by MTCam** [comment]
> > Should we concern ourselves with that before we are ready to use all array contexts?
> 
> If this PR supersedes #7, then I think yes.

Are all of your changes compatible with all array contexts?  If so - then I think we are good-to-go.

When #9 merges - this one will become the new Lazy Nozzle effort.  And feel free to push to this as needed, I was just trying to get #7 up-to-date with current driver and mirgecom@y1-production.

**Comment by MTCam** [comment]
> > Should we concern ourselves with that before we are ready to use all array contexts?
> 
> If this PR supersedes #7, then I think yes.

Also, Kaushik, we don't *need* to take over #7.  We can keep that one going and get rid of this one if we must merge this to #7.

**Comment by kaushikcfd** [comment]
Thanks @MTCam, yep these ops should be compatible with all array contexts. (the CI's happy too!).

> We can keep that one going and get rid of this one if we must merge this to #7.

I agree, it's probably better for the reviewers to keep these PRs separate.

File: nozzle.py [hunk 1]
```diff
@@ -516,9 +521,11 @@ def gen_sponge():
         amplitude = 1.0/current_dt/25.0
         x0 = 0.05
 
-        return (amplitude * actx.np.where(
-            nodes[0] > x0, zeros + ((nodes[0] - x0) / thickness) *
-            ((nodes[0] - x0) / thickness), zeros + 0.0))
+        return amplitude * actx.np.where(
+            actx.np.greater(nodes[-1], x0),
```
Comment (kaushikcfd) [inline nozzle.py:525]:
```suggestion
            actx.np.greater(nodes[0], x0),
```
Whoops this looks like a typo.

**Comment by MTCam** [comment]
@kaushikcfd I am unsure why your #7 closed - but your changes should be in this one too.   Do you want to take over here, or pull it into your own PR?  Are there further changes that you foresee for making nozzle lazy-ready?

**Comment by kaushikcfd** [comment]
@MTCam: Since this PR contains all the changes to make the array operations in the nozzle driver compatible with all the array contexts, I think it's fair that #7 got closed.

**Pull request merged**